casper/src/main/scala/coop/rchain/casper/Casper.scala

```diff
@@ -264,12 +272,143 @@ sealed abstract class MultiParentCasperInstances {
         if (isNotEquivocation) {
           Applicative[F].pure(Right(Valid))
         } else if (awaitingJustificationToChild.contains(block.blockHash)) {
-          Applicative[F].pure(Right(IncludeableEquivocation))
+          Applicative[F].pure(Right(AdmissibleEquivocation))
         } else {
           Applicative[F].pure(Left(IgnorableEquivocation))
         }
       }
 
+      private def neglectedEquivocationsCheckWithRecordUpdate(
+          block: BlockMessage,
+          dag: BlockDag): F[Either[RejectableBlock, IncludeableBlock]] =
+        if (equivocationsTracker.isEmpty) {
+          Applicative[F].pure(Right(Valid))
+        } else {
+          val neglectedEquivocationDetected =
+            equivocationsTracker.foldLeft(false) {
+              case (acc, equivocationRecord) =>
+                getEquivocationDiscoveryStatus(block,
+                                               dag,
+                                               equivocationRecord,
+                                               Set.empty[BlockMessage]) match {
+                  case EquivocationNeglected =>
+                    true
+                  case EquivocationDetected =>
+                    val updatedEquivocationDetectedBlockHashes = equivocationRecord.equivocationDetectedBlockHashes + block.blockHash
+                    equivocationsTracker.remove(equivocationRecord)
+                    equivocationsTracker.add(
+                      equivocationRecord.copy(
+                        equivocationDetectedBlockHashes = updatedEquivocationDetectedBlockHashes))
+                    acc
+                  case EquivocationOblivious =>
+                    acc
+                }
+
+            }
+          if (neglectedEquivocationDetected) {
+            Applicative[F].pure(Left(NeglectedEquivocation))
+          } else {
+            Applicative[F].pure(Right(Valid))
+          }
+        }
+
+      private def getEquivocationDiscoveryStatus(
+          block: BlockMessage,
+          dag: BlockDag,
+          equivocationRecord: EquivocationRecord,
+          equivocationChild: Set[BlockMessage]): EquivocationDiscoveryStatus = {
+        val equivocatingValidator = equivocationRecord.equivocator
+        val latestMessages        = toLatestMessages(block.justifications)
+        if (equivocationDetectable(latestMessages.toSeq,
+                                   dag,
+                                   equivocationRecord,
+                                   equivocationChild)) {
+          if (latestMessages.contains(equivocatingValidator)) {
```

Comment:
Maybe add a comment about how this works because we will have an upstream check for bond-justification correspondence.